common/path.go

```diff
@@ -100,14 +100,28 @@ func DefaultAssetPath() string {
 }
 
 func DefaultDataDir() string {
-	usr, _ := user.Current()
-	if runtime.GOOS == "darwin" {
-		return filepath.Join(usr.HomeDir, "Library", "Ethereum")
-	} else if runtime.GOOS == "windows" {
-		return filepath.Join(usr.HomeDir, "AppData", "Roaming", "Ethereum")
-	} else {
-		return filepath.Join(usr.HomeDir, ".ethereum")
+	// Try to place the data folder in the user's home dir
+	if usr, err := user.Current(); err == nil {
+		if runtime.GOOS == "darwin" {
+			return filepath.Join(usr.HomeDir, "Library", "Ethereum")
+		} else if runtime.GOOS == "windows" {
+			return filepath.Join(usr.HomeDir, "AppData", "Roaming", "Ethereum")
+		} else {
+			return filepath.Join(usr.HomeDir, ".ethereum")
+		}
+	}
+	// If the user cannot be retrieved (Android, Fedora/Go bug), use the current folder
+	if dir, err := os.Getwd(); err == nil {
+		if runtime.GOOS == "darwin" {
+			return filepath.Join(dir, "Ethereum")
+		} else if runtime.GOOS == "windows" {
+			return filepath.Join(dir, "Ethereum")
+		} else {
```

Comment:
I'm not sure about this. On android this will probably be fine. On Fedora it is a bug and when it gets fixed the datadir might change. And each time the user starts geth in a different working dir the datadir is different. This can get really confusing. Maybe its best on Fedora to exit the program and print a message informing the user that he needs to specify the datadir manually.